providers/mailjet/src/lib/mailjet.provider.ts

```diff
@@ -26,24 +25,28 @@ export class MailjetEmailProvider implements IEmailProvider {
       senderName: string;
     }
   ) {
-    this.mailjetClient = Client.connect(config.apiKey, config.apiSecret);
+    this.mailjetClient = new Mailjet.Client({
+      apiKey: config.apiKey,
+      apiSecret: config.apiSecret,
+    });
   }
 
   async sendMessage(
     emailOptions: IEmailOptions
   ): Promise<ISendMessageSuccessResponse> {
-    const send = this.mailjetClient.post('send', {
-      version: MAILJET_API_VERSION,
-    });
-    const requestObject = this.createMailData(emailOptions);
+    const response = await this.mailjetClient
+      .post('send', {
+        version: MAILJET_API_VERSION,
+      })
+      .request<Mailjet.SendEmailV3_1.Response>(
+        this.createMailData(emailOptions)
+      );
 
-    const response = (await send.request(
-      requestObject
-    )) as unknown as MailjetResponse;
+    const { body, response: clientResponse } = response;
 
     return {
-      id: response.response.header['x-mj-request-guid'],
-      date: response.response.header.date,
+      id: clientResponse.headers['x-mj-request-guid'],
+      date: new Date().toISOString(),
```

Comment:
i guess we are missing the provider date

Reply:
Yep :(